chainstate/src/rpc/types/output.rs

```diff
@@ -0,0 +1,141 @@
+
+use common::{
+    address::{AddressError, RpcAddress},
+    chain::{
+        output_value::OutputValue, stakelock::StakePoolData, timelock::OutputTimeLock,
+        tokens::TokenId, ChainConfig, DelegationId, Destination, PoolId, TxOutput,
+    },
+    primitives::amount::RpcAmountOut,
+};
+use crypto::vrf::VRFPublicKey;
+
+use super::token::{RpcNftIssuance, RpcTokenIssuance};
+
+#[derive(Debug, Clone, serde::Serialize)]
+pub enum RpcOutputValue {
+    Coin(RpcAmountOut),
+    Token(RpcAddress<TokenId>, RpcAmountOut),
+}
+
+impl RpcOutputValue {
+    fn new(chain_config: &ChainConfig, value: OutputValue) -> Result<Self, AddressError> {
+        let result = match value {
+            OutputValue::Coin(amount) => RpcOutputValue::Coin(RpcAmountOut::from_amount(
+                amount,
+                chain_config.coin_decimals(),
+            )),
+            OutputValue::TokenV0(_) => unimplemented!(),
+            OutputValue::TokenV1(token_id, amount) => RpcOutputValue::Token(
+                RpcAddress::new(chain_config, token_id)?,
+                RpcAmountOut::from_amount(amount, chain_config.coin_decimals()),
+            ),
+        };
+        Ok(result)
+    }
+}
+
+#[derive(Debug, Clone, serde::Serialize)]
+pub struct RpcStakePoolData {
+    pledge: RpcAmountOut,
+    staker: RpcAddress<Destination>,
+    vrf_public_key: RpcAddress<VRFPublicKey>,
+    decommission_key: RpcAddress<Destination>,
+    margin_ratio_per_thousand: String,
+    cost_per_block: RpcAmountOut,
+}
+
+impl RpcStakePoolData {
+    fn new(chain_config: &ChainConfig, data: &StakePoolData) -> Result<Self, AddressError> {
+        let result = Self {
+            pledge: RpcAmountOut::from_amount(data.pledge(), chain_config.coin_decimals()),
+            staker: RpcAddress::new(chain_config, data.staker().clone())?,
+            vrf_public_key: RpcAddress::new(chain_config, data.vrf_public_key().clone())?,
+            decommission_key: RpcAddress::new(chain_config, data.decommission_key().clone())?,
+            margin_ratio_per_thousand: data.margin_ratio_per_thousand().into_percentage_str(),
+            cost_per_block: RpcAmountOut::from_amount(
+                data.cost_per_block(),
+                chain_config.coin_decimals(),
+            ),
+        };
+        Ok(result)
+    }
+}
+
+#[derive(Debug, Clone, serde::Serialize)]
+pub enum RpcTxOutput {
+    Transfer(RpcOutputValue, RpcAddress<Destination>),
+    LockThenTransfer(RpcOutputValue, RpcAddress<Destination>, OutputTimeLock),
+    Burn(RpcOutputValue),
+    CreateStakePool(RpcAddress<PoolId>, Box<RpcStakePoolData>),
+    ProduceBlockFromStake(RpcAddress<Destination>, RpcAddress<PoolId>),
+    CreateDelegationId(RpcAddress<Destination>, RpcAddress<PoolId>),
+    DelegateStaking(RpcAmountOut, RpcAddress<DelegationId>),
+    IssueFungibleToken(Box<RpcTokenIssuance>),
+    IssueNft(
+        RpcAddress<TokenId>,
+        Box<RpcNftIssuance>,
+        RpcAddress<Destination>,
+    ),
```

Comment:
These translate into json arrays. That's not good. Please use named content. So instead of:

```
Transfer(RpcOutputValue, RpcAddress<Destination>)
```

make it
```
Transfer({
    value: RpcOutputValue,
    destination: RpcAddress<Destination>,
})
```